test/test-template-engine.js

```diff
@@ -1094,80 +1094,70 @@ describe('value issues', () => {
 
 describe('html errors', () => {
   it('throws when attempting to interpolate within a style tag', () => {
-    const container = document.createElement('div');
-    const callback = () => render(container, html`
+    const callback = () => html`
       <style>
         div { background-color: ${'red'}; }
       </style>
-    `);
+    `;
     const expectedMessage = 'Interpolation of "style" tags is not allowed.';
     assertThrows(callback, expectedMessage);
   });
 
   it('throws when attempting to interpolate within a script tag', () => {
     const evil = '\' + prompt(\'evil\') + \'';
-    const container = document.createElement('div');
-    const callback = () => render(container, html`
+    const callback = () => html`
       <script id="target">
         console.log('${evil}');
       </script>
-    `);
+    `;
     const expectedMessage = 'Interpolation of "script" tags is not allowed.';
     assertThrows(callback, expectedMessage);
   });
 
   it('throws when attempting non-trivial interpolation of a textarea tag', () => {
-    const container = document.createElement('div');
-    const callback = () => render(container, html`<textarea id="target">please ${'foo'} no</textarea>`);
+    const callback = () => html`<textarea id="target">please ${'foo'} no</textarea>`;
     const expectedMessage = 'Only basic interpolation of "textarea" tags is allowed.';
     assertThrows(callback, expectedMessage);
   });
 
   it('throws when attempting non-trivial interpolation of a textarea tag via nesting', () => {
-    const container = document.createElement('div');
-    const callback = () => render(container, html`<textarea id="target"><b>please ${'foo'} no</b></textarea>`);
+    const callback = () => html`<textarea id="target"><b>please ${'foo'} no</b></textarea>`;
     const expectedMessage = 'Only basic interpolation of "textarea" tags is allowed.';
     assertThrows(callback, expectedMessage);
   });
 
   it('throws when attempting non-trivial interpolation of a title tag', () => {
-    const container = document.createElement('div');
-    const callback = () => render(container, html`<title>please ${'foo'} no</title>`);
+    const callback = () => html`<title>please ${'foo'} no</title>`;
```

Comment:
Note that we no longer need to `render` to see an error. This separation of concerns feels appropriate… but TBD on whether this would ever be a performance bottleneck.